master/internal/checkpoints/postgres_checkpoints.go

```diff
@@ -27,11 +30,11 @@ func (db *PgDB) CheckpointByUUID(id uuid.UUID) (*model.Checkpoint, error) {
 }
 
 // CheckpointByUUIDs looks up a checkpoint by list of UUIDS, returning nil if error.
-func (db *PgDB) CheckpointByUUIDs(ckptUUIDs []uuid.UUID) ([]model.Checkpoint, error) {
+func CheckpointByUUIDs(ctx context.Context, ckptUUIDs []uuid.UUID) ([]model.Checkpoint, error) {
```

Comment:
question (non-blocking): why do we have both `CheckpointByUUIDs()` and `CheckpointByUUID()`? 

Is the query planner/optimizer in Postgres not able to simplify a single element array IN where clause into an equality check?

Reply:
Uh I'm not sure why we have two. I do know that CheckpointByUUID is not supposed to return an error if the checkpoint is not found but CheckpointByUUIDs returns errors if checkpoints aren't found

Reply:
Good point, I haven't looked too deeply into how these are being used, I'm also curious why we expect different error behavior between these two; seems like it could led to some gotcha bugs